cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/BrokerStateProvider.java

```diff
@@ -0,0 +1,122 @@
+        return new HttpClientOptions()
+                .setLogActivity(true)
+                .setSsl(true)
+                .setVerifyHost(true)
+                .setPemTrustOptions(pto)
+                .setKeyCertOptions(pkco);
+    }
+
+    protected <T> Future<String> doGet(String host) {
+        LOGGER.debugCr(reconciliation, "Making GET request to {}", BROKER_STATE_REST_PATH);
+        return HttpClientUtils.withHttpClient(vertx, getHttpClientOptions(), (httpClient, result) ->
+                httpClient.request(HttpMethod.GET, BROKER_STATE_HTTPS_PORT, host, BROKER_STATE_REST_PATH, request -> {
+                    if (request.succeeded()) {
+                        request.result().setFollowRedirects(true)
+                        .putHeader("Accept", "application/json");
+                        request.result().send(response -> {
+                            if (response.succeeded()) {
+                                if (response.result().statusCode() == 200) {
+                                    response.result().bodyHandler(body -> {
+                                        var jsonBody = body.toString();
+                                        result.complete(jsonBody);
+                                    });
+                                } else {
+                                    result.fail(new RuntimeException("Unexpected HTTP status code: " + response.result().statusCode()));
+                                }
+                            } else {
+                                result.tryFail(response.cause());
+                            }
+                        });
+                    } else {
+                        result.tryFail(request.cause());
+                    }
+                }));
+    }
+
+    public BrokerState getBrokerState(String podName) {
+        BrokerState brokerstate = new BrokerState(-1, null);
+        String host = DnsNameGenerator.podDnsName(namespace, KafkaResources.brokersServiceName(cluster), podName);
+        CompletableFuture<String> completableFuture = new CompletableFuture<>();
+
+        Future<String> getFuture = doGet(host);
+        getFuture.onComplete(future -> {
+            if (future.succeeded()) {
+                completableFuture.complete(future.result());
+            } else {
+                completableFuture.completeExceptionally(future.cause());
+            }
+        });
+
+        try {
+            var result = completableFuture.get(3, TimeUnit.SECONDS);
+            ObjectMapper mapper = new ObjectMapper();
+            brokerstate = mapper.readValue(result, BrokerState.class);
+        } catch (TimeoutException | InterruptedException | ExecutionException e) {
+            LOGGER.warnCr(reconciliation, "Failed to get broker state", e);
+        } catch (JsonProcessingException jsonException) {
+            LOGGER.warnCr(reconciliation, "Failed to parse response for broker state", jsonException);
+        }
```

Comment:
If you're returning `new BrokerState(-1, null)` in the event of these errors you should at least state that in the javadoc of this method, since it's not obvious that that would be the contract.